### ISSUE TITLE
Update for the visibility of the game to unregistered user

### ISSUE THREAD

**Issue by frknlke** [bounswe/bounswe2023group7#529]
### Issue Description

In our mobile app, unregistered users can neither view the games nor create  a new game page. We, mobile and front-end teams, discussed about this subject and decided to change its functionality so that unregistered users could view the games but could not create a game. I will update necessary places in user session mechanism to dictate this rule.

### Tasks

Parts of the task:
- [x] Change rules to view games page
- [x] Change rules to create a new game page

### Extra Comments

None

### Final Step

None

### Deadline of the Issue

28.11.2023-Tuesday-10:00

### Reviewer

@haticerk

### Deadline for the Review

28.11.2023-Tuesday-13:00

**Comment by frknlke** [comment]
I changed the rules for accessing games page. Now unregistered users will be able to scroll through games page too.

**Comment by frknlke** [comment]
I've added a new rule to restrict access to the game creation page, so now only logged in users can create a new game on the games page.

**Comment by frknlke** [comment]
I've also added rule to regulate accesses made in game profile page. Now only logged in users can follow or unfollow games. Unlogged users view just follow button. If they pressed on the button, an informative message will be popped up to prompt users that they must log in if they want to follow a game. I added this informative snackbar also to the unauthorized accesses made to reach create games page.
